src/pages/EventsPage.js

```diff
@@ -54,52 +54,55 @@ const useStyles = makeStyles(theme => ({
   },
 }));
 
-export default function EventPage(props) {
+const getEvents = async ecat => {
+  try {
+    const url = ecat !== undefined ? `${hostname}/api/event/cat/${ecat}` : `${hostname}/api/event`;
+    const response = await axios.get(url);
+    const events = response.data.events;
+    events.forEach(event => (event.eventstart = new Date(event.eventstart)));
+    return events.sort((a, b) => b.eventstart - a.eventstart);
+  } catch (error) {
+    console.error(error);
+    throw new Error('Failed to fetch events');
+  }
+};
+
+export default function EventPage() {
   const classes = useStyles();
-  const [loading, setLoading] = React.useState(false);
+  const { search } = useLocation();
+  const { ecat } = queryString.parse(search);
   let loggedIn = localStorage.getItem('isAuthenticated') === 'true';
-  const { ecat } = queryString.parse(props.location.search);
+  // Get events from db and use a list to manage which events are being shown
+  const { data: events, isFetching, error } = useQuery(['events', ecat], () => getEvents(ecat));
 
-  //Get events from db and use a list to manage which events are being shown
-  const [events, setEvents] = useState([]);
-  const [list, setList] = React.useState([
-    { hi: 'these' },
-    { dont: 'mean' },
-    { anything: 'they' },
-    { help: 'skeletons' },
-  ]);
+  // Keyword search
+  const [text, setText] = useState('');
+  const [filteredEvents, setFilteredEvents] = useState([]);
 
-  //search text
-  const [text, setText] = React.useState('');
-  //Keyword search
   const handleSearch = event => {
-    setText(event.target.value);
+    const searchText = event.target.value;
+    setText(searchText);
+    if (searchText === '') {
+      setFilteredEvents([]);
+    } else {
+      const filteredEvents = events.filter(event => {
+        const titleMatch = event.ename.toLowerCase().includes(text.toLowerCase());
+        const descriptionMatch = event.details.toLowerCase().includes(text.toLowerCase());
+        return titleMatch || descriptionMatch;
+      });
+      setFilteredEvents(filteredEvents);
+    }
   };
 
-  useEffect(() => {
-    var updatedList = events;
-    updatedList = events.filter(function (item) {
-      if (text === '') return events;
-      return (
-        item.keywords.toLowerCase().search(text.toLowerCase()) !== -1 ||
-        item.ename.toLowerCase().search(text.toLowerCase()) !== -1
-      );
-    });
-    setList(updatedList);
-  }, [events, text]);
-
-  //Society names for dropdown filter
-  const [category, setCategory] = React.useState(ecat !== undefined ? ecat : 0);
-  //Society filter
+  const [category, setCategory] = useState(ecat ? ecat : 0);
```

Comment:
You seem to want to do coalescing. This may be more compact
```suggestion
  const [category, setCategory] = useState(ecat ?? 0);
```